### PR TITLE
Prepare project for semi-pinned tags for major version

### PR DESCRIPTION
**Pull request by salome-voltz**
This MR aims to prepare this project for semi-pinned tags for major versions. 
Tags will be created and updated in ggshield-tools release script. 

This MR needs to be merged first.
Works with MR#22 on ggshield-tools. Needs to be merged first.

### CODE REVIEW

File: .github/workflows/tag.yml [hunk 1]
```diff
@@ -3,13 +3,13 @@ name: Application Release
 on:
   push:
     tags:
-      - "v*"
+      - "v1.*"
```
Comment (agateau-gg) [inline .github/workflows/tag.yml:6]: Do we need this change? If we ever make a v2 we would still need a similar CI workflow, no?

**Review by agateau-gg — CHANGES_REQUESTED** [review]

File: .github/workflows/tag.yml (hunk 1 above)
Comment (salome-voltz) [reply .github/workflows/tag.yml]: We need to distinguish "v1" tag from version tags (like v1.20.0) here to make sure that we don't trigger a release when we push "v1" tag to its new position. I changed it to "v*.*", wdyt ?

Comment (agateau-gg) [reply .github/workflows/tag.yml]: Oh right, good catch! "v*.*" sounds good. Can you add a comment about this?

Comment (salome-voltz) [reply .github/workflows/tag.yml]: Yes ! :smile:

**Review by agateau-gg — APPROVED** [review]

**Pull request merged**